src/query/catalog/src/plan/datasource/datasource_info/mod.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_expression::TableSchema;
+use common_meta_app::schema::TableInfo;
+
+mod parquet;
+mod parquet_read_options;
+
+mod stage;
+
+pub use parquet::ParquetTableInfo;
+pub use parquet_read_options::ParquetReadOptions;
+pub use stage::StageTableInfo;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum DataSourceInfo {
+    // Normal table source, `fuse/system`.
+    TableSource(TableInfo),
+    // Internal/External source, like `s3://` or `azblob://`.
+    StageSource(StageTableInfo),
+    // stage source with parquet format used for select.
+    ParquetSource(ParquetTableInfo),
```

Comment:
The abstraction is a bit confusing to me. `ParquetSource` is on a stage too, right?

The difference from `StageSource` and `ParquetSource` is `StageSource` used the table meta provided by metasrv, and `ParquetSource` has it's own table meta (inferred from files).

Do we have better namings?

Reply:
StageSource -> StageTable (impl Table)
ParquetSource -> ParquetTable (impl Table)


we can refactor it after

1. read_parquet is removed, (get rid of some limits and redundancy)
2. add `select from csv/tsv` (the relation will be more clear)

Reply:
Mark as resolved